libs/build_gsl.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -eux
+
+name="gsl"
+version=${1:-${STACK_gsl_version}}
+
+compiler=$(echo $HPC_COMPILER | sed 's:/:-:g')
```

Comment:
```suggestion
compiler=$(echo $HPC_COMPILER | sed 's/\//-/g')
```

Reply:
Isn't it a lot cleaner without escaping the forward slash?

Reply:
Yes. It is a lot cleaner. 
I was going for consistency, but we can go forth with this from now on.